File: common/internal/src/main/java/dcp/mc/pstp/config/EffectsProtection.java
package dcp.mc.pstp.config;

import org.jetbrains.annotations.ApiStatus.Internal;
import org.jetbrains.annotations.NotNull;

@Internal
public record EffectsProtection(
        boolean enabled,
        @NotNull String @NotNull [] living,
        @NotNull String @NotNull [] undead
)
{
    EffectsProtection() {
        this(true, new String[] { "minecraft:instant_damage", "minecraft:poison" }, new String[] { "minecraft:instant_health", "minecraft:regeneration" });
    }
}

File: common/src/main/kotlin/dcp/mc/pstp/config/EffectsProtection.kt
package dcp.mc.pstp.config

@JvmRecord
data class EffectsProtection (
    val enabled: Boolean = true,
    val livingEffects: List<String> = listOf("minecraft:instant_damage", "minecraft:poison"),
    val undeadEffects: List<String> = listOf("minecraft:instant_health", "minecraft:regeneration")
)